### PR TITLE
unity: add support for xenial

### PR DESCRIPTION
**Pull request by smibarber**
Also retire unity hacks for utopic and other unsupported releases.

### CODE REVIEW

File: chroot-bin/crouton-unity-autostart [hunk 1]
```diff
@@ -17,9 +17,9 @@ execfromconf() {
     fi
 }
 
-# Upstart launches unity on utopic, and this needs to be started before
+# Upstart launches unity on utopic+, and this needs to be started before
 # any dependent services.
-if [ "$RELEASE" = 'utopic' ]; then
+if [ "$RELEASE" = 'xenial' ]; then
```
Comment (dnschneid) [inline chroot-bin/crouton-unity-autostart]: Maybe throw yakkety in there too?

File: chroot-bin/gnome-session-wrapper [hunk 2]
```diff
@@ -33,6 +33,10 @@ if [ "${DESKTOP_SESSION#cinnamon}" != "$DESKTOP_SESSION" ] && \
     SESSION="$CINNAMON_SESSION"
 fi
 
+SESSION_FILE="/usr/share/$SESSION/sessions/$DESKTOP_SESSION.session"
+XDG_CURRENT_DESKTOP=`awk -F "=" '/DesktopName/ {print $2}' "$SESSION_FILE"`
```
Comment (dnschneid) [inline chroot-bin/gnome-session-wrapper]: quotes, and use `$( )`

File: chroot-bin/gnome-session-wrapper [hunk 3]
```diff
@@ -33,6 +33,10 @@ if [ "${DESKTOP_SESSION#cinnamon}" != "$DESKTOP_SESSION" ] && \
     SESSION="$CINNAMON_SESSION"
 fi
 
+SESSION_FILE="/usr/share/$SESSION/sessions/$DESKTOP_SESSION.session"
```
Comment (dnschneid) [inline chroot-bin/gnome-session-wrapper]: does this session file exist on other releases?

Comment (smibarber) [inline chroot-bin/gnome-session-wrapper]: Yep, for any combination of GNOME/Unity (gnome-session) or cinnamon (cinnamon-session).

File: chroot-bin/crouton-unity-autostart (hunk 1 above)
Comment (smibarber) [inline chroot-bin/crouton-unity-autostart]: Yakkety has more strange unity breakage. I'd say let's just get the LTS working first.

File: chroot-bin/gnome-session-wrapper (hunk 2 above)
Comment (smibarber) [inline chroot-bin/gnome-session-wrapper]: Done.

File: targets/x11-common [hunk 4]
```diff
@@ -50,7 +50,9 @@ echo 'mode: blank' > /etc/skel/.xscreensaver
 mkdir -p /usr/share/desktop-directories
 
 # Prevent Upstart from taking over X sessions
-echo > /etc/upstart-xsessions
+if release -ge trusty -le xenial; then
```
Comment (dnschneid) [inline targets/x11-common]: all conditions are OR'd together.

`if release -ge trusty && release -le xenial; then`

or

`if release -eq trusty -eq xenial; then`

File: targets/x11-common [hunk 5]
```diff
@@ -50,7 +50,9 @@ echo 'mode: blank' > /etc/skel/.xscreensaver
...
+    dpkg-divert --local --rename --add /etc/upstart-xsessions
```
Comment (dnschneid) [inline targets/x11-common]: is this okay if you run it multiple times?

**Pull request merged**